src/main.rs

```diff
@@ -0,0 +1,65 @@
+use std::io;
+
+struct RomanVal<'a> {
+    val: i32,
+    roman: &'a str,
+}
+
+fn main() {
+    println!("Enter a number");
+
+    let mut input = String::new();
+
+    io::stdin().read_line(&mut input)
+        .expect("Failed to read line");
+
+    let input = input.trim();
+    let num: i32 = input.parse().unwrap();
+
+    println!("You entered '{}'", num);
+
+    let roman = to_roman(num);
+    println!("That's '{}' in Roman Numerals", roman);
+}
+
+fn to_roman(n: i32) -> String {
+    if n > 3999 || n <= 0 {
+        // better to return a Result enum?
+        panic!("number must be between 1 and 3999, cus I don't want to code stuff like V̅ etc")
+    }
```

Comment:
I think so, if you were building a proper library a panic wouldn't be that helpful. Yeah?

`Result<String, ErrorCodes>`

with like

`Err(ErrorCodes::OutsideBoundaries)`